plugins/field-slider/test/field_slider_test_v2.mocha.js

```diff
@@ -0,0 +1,119 @@
+    {title: 'Integer', value: 1, expectedValue: 1},
+    {title: 'Float', value: 1.5, expectedValue: 1.5},
+    {title: 'Integer String', value: '1', expectedValue: 1},
+    {title: 'Float String', value: '1.5', expectedValue: 1.5},
+    {title: 'Infinity', value: Infinity, expectedValue: Infinity},
+    {title: 'Negative Infinity', value: -Infinity, expectedValue: -Infinity},
+    {title: 'Infinity String', value: 'Infinity', expectedValue: Infinity},
+    {title: 'Negative Infinity String', value: '-Infinity',
+      expectedValue: -Infinity},
+  ];
+
+  suite('Constructor', function() {
+    test('Empty', function() {
+      const field = new FieldSlider();
+      helpers.assertSliderFieldDefault(field);
+    });
+    runs.forEach(function(run) {
+      // Monica's note: My understanding is that by having the if statement out
+      // of the test function, the check doesn't happen when the test is run.
+      if (run.expectDefault) {
+        test(run.title, function() {
+          const field =
+              helpers.createSliderFieldSameValuesConstructor(run.value);
+          helpers.assertSliderFieldDefault(field);
+        });
+      } else {
+        test(run.title, function() {
+          const field =
+              helpers.createSliderFieldSameValuesConstructor(run.value);
+          helpers.assertSliderFieldSameValues(field, run.expectedValue);
+        });
+      }
+    });
+  });
+  suite('fromJson', function() {
+    test('Empty', function() {
+      const field =FieldSlider.fromJson({});
+      helpers.assertSliderFieldDefault(field);
+    });
+    runs.forEach(function(run) {
+      if (run.expectDefault) {
+        test(run.title, function() {
+          const field = helpers.createSliderFieldSameValuesJson(run.value);
+          helpers.assertSliderFieldDefault(field);
+        });
+      } else {
+        test(run.title, function() {
+          const field = helpers.createSliderFieldSameValuesJson(run.value);
+          helpers.assertSliderFieldSameValues(field, run.expectedValue);
+        });
+      }
+    });
+  });
+  suite('setValue', function() {
+    suite('Empty -> New Value', function() {
+      setup(function() {
+        this.field = new FieldSlider();
+      });
+      runs.forEach(function(run) {
+        if (run.expectDefault) {
+          test(run.title, function() {
+            this.field.setValue(run.value);
+            helpers.assertFieldValueDefault(this.field);
+          });
+        } else {
+          test(run.title, function() {
+            this.field.setValue(run.value);
+            helpers.assertFieldValue(this.field, run.expectedValue);
+          });
+        }
+      });
+    });
+    suite('Vale -> New Value', function() {
+      setup(function() {
+        this.field = new FieldSlider(1);
+      });
+      runs.forEach(function(run) {
```

Comment:
A helper method that abstracts all this away into a simple ``runTests`` could help with readability.
eg: 
```
runTests(() => {
   ... 
});
```